### DISCUSSION TITLE
Permission denied to create file in "/github/home/" folder

### DISCUSSION THREAD

**Discussion by epomatti** [community/community#25841] [Actions]
<p>I have this simple workflow to build a Flutter app, but it is failing to create a file at the home <strong>/github/home/</strong> folder.</p>
<pre><code>Run flutter pub get

FileSystemException: Cannot create file, path = '/github/home/.flutter' (OS Error: Permission denied, errno = 13)
#0 _File.throwIfError (dart:io/file_impl.dart:643:7)
#1 _File.createSync (dart:io/file_impl.dart:281:5)
</code></pre>
<p>I attached the container initialization log and it is possible to see that the docker image has the home volume mapped to it.</p>
<pre><code>name: Dart CI

on: [push]

jobs:
  build:

    runs-on: ubuntu-latest

    container:
      image: cirrusci/flutter:v1.7.8-hotfix.4

    steps:
    - uses: actions/checkout@v1
    - name: Install dependencies
      run: flutter pub get
      working-directory: my_app
    - name: Run tests
      run: flutter test
      working-directory: my_app
</code></pre>
<p>Here are the logs for this action.</p>
<p><a href="https://github.com/actions/docker/files/3587247/initialize-containers.log" rel="nofollow noopener">https://github.com/actions/docker/files/3587247/initialize-containers.log</a></p>
<p><a href="https://github.com/actions/docker/files/3587248/install-dependencies.log" rel="nofollow noopener">https://github.com/actions/docker/files/3587248/install-dependencies.log</a></p>

**Comment by ethomson** [comment]
<p>I’m not familiar with the <code>cirrusci/flutter</code> container.  What user is that running as?  You may need to run some commands in that container as root.</p>

**Comment by Sunbreak** [comment]
<p><code>flutter pub get</code> will create <code>~/.pub-cache</code></p>
<p>It works on Gitlab CI with the same image, but doesn’t on GitHub Action</p>

**Comment by epomatti** [comment, marked as answer]
<p>Solved it by adding <strong>“sudo”</strong> before any command that requires permission. It also requires that the full path of Flutter is referenced when executing the commands.</p>
<p>The image was running with “cirrus” user and therefore didn’t have permission.</p>
<p>More info here: <a href="https://stackoverflow.com/a/58469603/3231778" rel="nofollow noopener">https://stackoverflow.com/a/58469603/3231778</a></p>
